sunpy/instr/goes.py

```diff
@@ -583,16 +605,490 @@ def _goes_get_chianti_em(longflux, temp, satellite=8, abundances="coronal",
     modelflux = np.asarray(modelflux)
 
     # Ensure input values of flux ratio are within limits of model table
-    if np.min(np.log10(temp)) < np.min(modeltemp) or \
-      np.max(np.log10(temp)) > np.max(modeltemp) or \
-      np.isnan(np.min(np.log10(temp))):
+    if np.min(np.log10(temp.value)) < np.min(modeltemp) or \
+      np.max(np.log10(temp.value)) > np.max(modeltemp) or \
+      np.isnan(np.min(np.log10(temp.value))):
         raise ValueError("All values in temp must be within the range "
                          "{0} - {1} MK.".format(np.min(10**modeltemp),
                                                 np.max(10**modeltemp)))
 
     # Perform spline fit to model data
     spline = interpolate.splrep(modeltemp, modelflux, s=0)
-    denom = interpolate.splev(np.log10(temp), spline, der=0)
-    em = longflux/denom * 1e55
+    denom = interpolate.splev(np.log10(temp.value), spline, der=0)
```

Comment:
Wouldn't be better to assign to a variable the value of `np.log10(temp.value)` - above the `if` - instead of calling it 4 times? Then, probably the if block would look better.